Hypeclass/Source/Schedule/ScheduleViewController.swift

```diff
@@ -190,31 +214,80 @@ class ScheduleViewController: BaseViewController {
             stackView.spacing = 20
             stackView.distribution = .equalSpacing
             
-            // TODO: Should be replaced to real data management!
-            for n in 0...3 {
-                // TODO: Fetch class information here
-                let dancerName = "NARAE"
-                let studioName = "1million"
-                let startTime = "18:00"
-                let endTime = "20:00"
+            weekSchedules[$0.0].forEach { schedule in
+                let dancerName = schedule.dancerName
+                let studioName = schedule.studioID
+                let startTime = "\(schedule.startTime.get(.hour)):\(schedule.startTime.get(.minute))"
+                let endTime = "\(schedule.endTime.get(.hour)):\(schedule.endTime.get(.minute))"
                 
                 let btn = ScheduleButton(frame: .zero, dancerName: dancerName, studioName: studioName, startTime: startTime, endTime: endTime)
-                btn.tag = n // TODO: DanceClass id를 Int로 변환
+                btn.tag = 0 // TODO: Dancer id를 Int로 변환
                 btn.addTarget(self, action: #selector(pushDetailView), for: .touchUpInside)
+                
                 stackView.addArrangedSubview(btn)
             }
         }
     }
     
     /// ScheduleViewController에서 변경되는 뷰를 다시 로드합니다.
-    func reloadViews() {
-        monthNumLabel.text = String(selectedDate.get(.month))
+    private func reloadViews() {
+        monthNumberLabel.text = monthString()
+        addButtonToStackView()
         scheduleCollectionView.reloadData()
         weekCollectionView.reloadData()
     }
+    
+    /// selectedDate와 같은 주의 월요일을 반환합니다.
+    private func mondayInWeek() -> Date {
+        let cal = Calendar.current
+        var date = selectedDate
+        let weekdayNum = date.get(.weekday)
+
+        if weekdayNum == 1 {
+            date = cal.date(byAdding: .day, value: -7, to: date)!
+        }
+
+        var comps = cal.dateComponents([.weekOfYear, .yearForWeekOfYear], from: date)
+        comps.weekday = 2
+        return cal.date(from: comps)!
+    }
```

Comment:
단순한 의견입니다만! 함수가 함수 바깥의 맥락을 많이 캡쳐하는 것으로 보입니다.
이들을 함수의 파라미터로 바꿔준다면, 함수의 역할을 이해하는 데 더 도움이 될것 같아요! 어떻게 생각하시나요 ㅎㅎ

Ex. cal, weekdayNum 등을 파라미터로 받는건 어떨까요? 아래는 제안입니다!
```suggestion
    private func mondayInWeek(at weekdayNum: Int, from calendar: ) -> Date {
        var date = selectedDate

        if weekdayNum == 1 {
            date = cal.date(byAdding: .day, value: -7, to: date)!
        }

        var comps = cal.dateComponents([.weekOfYear, .yearForWeekOfYear], from: date)
        comps.weekday = 2
        return cal.date(from: comps)!
    }
```

Reply:
Calendar.current 처리와 함수 내에서 바깥 맥락을 계속 사용하는 게 고민이었는데 파라미터로 넘기면 해결되네요! 좋은 리뷰 감사합니다! 반영하겠습니다 :)

Reply:
반영했습니다 ✅